python/cudf/cudf/dataframe/columnops.py

```diff
@@ -412,8 +412,7 @@ def as_column(arbitrary, nan_as_null=True, dtype=None, name=None):
                 "be typecast to a Date64 value",
                 UserWarning,
             )
-            arbitrary = arbitrary.cast(pa.date64())
-            data = as_column(arbitrary)
+            data = as_column(arbitrary.cast(pa.int32())).astype("M8[ms]")
```

Comment:
Instead of defaulting to milliseconds here should we use the least granular option we have which would be seconds?

Reply:
I kept this conversion to milliseconds since that's what it was doing before. Didn't figure we should change the behavior ingesting `pa.Date32` until we support 32bit unix days.

Reply:
I guess it really doesn't matter here. Can change if user's give feedback of wanting it changed :)